revapi-basic-features/src/main/java/org/revapi/basic/VersionsTransform.java

```diff
@@ -622,7 +621,16 @@ private static TextModification parseTextModification(JsonNode modificationNode)
                 return new TextModification(modificationNode.asText(), null, null);
             } else if (modificationNode.isObject()) {
                 String prepend = modificationNode.path("prepend").asText();
-                String append = modificationNode.path("append").asText();
+                String append = null;
+                if (modificationNode.path("append").path("majorIncreaseRequired").asBoolean()) {
```

Comment:
You define `majorIncreaseRequired` (and others) as `string` in the schema (which is a correct thing to do). Casting it here as a boolean doesn't make sense.

`TextModification` is basically a runtime representation of the configuration. So here we want to capture all the possibilities that can happen in the configuration so that later on, we can apply that configuration on the difference.

Reply:
Ya, I should have thought about possibilities that can happen in the configuration...
Can you please have a look now? I think it will cover all the possibilities that can happen in the configuration.